### ISSUE TITLE
"test -r expanded" display issue

### ISSUE THREAD

**Issue by ra1u** [dart-lang/test#1123]
# Environment

- `pub version` or `flutter pub version`: Pub 2.7.0
- OS version: Debian 10

# Problem
Issue with output running tests with flag -j > 1

command is `/usr/lib/dart/bin/pub run test -r expanded `
project and code is https://github.com/ra1u/redis-dart
(redis requred for running test, check https://github.com/ra1u/redis-dart/blob/master/test/main.dart for non default config)

## Expected behavior
 output as with -j 1

```
% /usr/lib/dart/bin/pub run test -r expanded -j 1
00:00 +0: test/cas_test.dart: Test Incr CAS Multiple
00:01 +1: test/error_test.dart: Throw  received Redis Errors Expect error when sending Garbage
00:01 +2: loading test/lua_test.dart
00:02 +2: test/lua_test.dart: Test Lua Native
00:02 +3: loading test/pubsub_test.dart
00:02 +3: test/pubsub_test.dart: Test Redis Pub-Sub Publishing to channel before subscription
00:02 +4: loading test/basic_test.dart
00:02 +4: test/pubsub_test.dart: Test Redis Pub-Sub Subscribe to channel
00:02 +5: loading test/basic_test.dart
00:02 +5: test/pubsub_test.dart: Test Redis Pub-Sub Publishing to channel
00:02 +6: loading test/basic_test.dart
00:02 +6: test/pubsub_test.dart: Test Redis Pub-Sub Unsubscribe channel
00:02 +7: loading test/basic_test.dart
00:03 +7: test/unicode_test.dart: Test Unicode Support Set and Get Unicode Value
00:03 +8: test/basic_test.dart: Basic Redis Functionality test One by One
00:03 +9: loading test/transactions_test.dart
00:04 +9: test/transactions_test.dart: Basic Transaction Test
00:04 +10: loading test/conversion_test.dart
00:04 +10: test/transactions_test.dart: Fake CAS Transaction Fake CAS
00:04 +11: loading test/conversion_test.dart
00:04 +11: test/transactions_test.dart: Fake CAS Transaction Fake CAS Multiple
00:04 +12: loading test/conversion_test.dart
00:04 +12: test/conversion_test.dart: Redis Type Conversion Expect Integer conversion to Bulk String
00:04 +13: All tests passed!
```

## Actual behavior

default is -j  4 per --help
```
/usr/lib/dart/bin/pub run test -r expanded     
00:00 +0: test/basic_test.dart: Basic Redis Functionality test One by One
00:00 +1: test/basic_test.dart: Basic Redis Functionality test One by One
00:00 +2: test/cas_test.dart: Test Incr CAS Multiple
00:00 +3: test/cas_test.dart: Test Incr CAS Multiple
00:00 +4: test/cas_test.dart: Test Incr CAS Multiple
00:00 +5: test/cas_test.dart: Test Incr CAS Multiple
00:00 +6: test/cas_test.dart: Test Incr CAS Multiple
00:00 +7: test/cas_test.dart: Test Incr CAS Multiple
00:00 +8: test/cas_test.dart: Test Incr CAS Multiple
00:00 +9: test/cas_test.dart: Test Incr CAS Multiple
00:00 +10: test/cas_test.dart: Test Incr CAS Multiple
00:00 +11: test/cas_test.dart: Test Incr CAS Multiple
00:00 +12: test/cas_test.dart: Test Incr CAS Multiple
00:01 +13: All tests passed!
```
Issue is that some lines are repeating and overlapping

**Comment by jakemac53** [comment]
This is likely a duplicate of https://github.com/dart-lang/test/issues/398.

I agree that for the expanded reporter at least it isn't really the expected behavior though. I think its worth considering changing this but I also think that it might require a significant rework to do so.

**Comment by jakemac53** [comment]
tldr; It prints the oldest running test after any test completes not the test that just completed - because that was determined to be more useful.

**Comment by jakemac53** [comment]
An alternative proposal to the current behavior could be to use a heartbeat and print periodically the list of tests which have been running for more than some threshold of time. This would have the same benefits of showing you what are the slow/hung tests, while not having confusing output in expanded mode.

**Comment by jakemac53** [comment]
cc @natebosch @grouma

**Comment by ra1u** [comment]
I dont  I need to run this particular options. In general I would find  some sort of report that shows what tests are running and what is its result highly beneficial.

**Comment by jakemac53** [comment]
Note that you can also use the json reporter and create your own view of the results should you choose to do so - that is documented here https://github.com/dart-lang/test/blob/ea37dba336b6525d7e6489105b4858c9279c43f3/pkgs/test/doc/json_reporter.md.

That gives a lot of fine grained information.

It is somewhat unlikely that we can change how any of the other reporters work in any fundamental way since it could break other tools.

**Comment by sebthom** [comment]
Six years later the issue is still present.

I get dozen of repeated output lines like this:
```
00:00 +29: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +30: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +31: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +32: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +33: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +34: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +35: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +36: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +37: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +38: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
00:00 +39: test\serializers\json_seq_serializer_test.dart: JsonSeqSerializer check registration
```

Each line stands for a different test but the wrong/same title is reported.

only way to solve it is by using concurrency 1, e.g. `dart test -r expanded --concurrency`
